src/Gaffer/MetadataAlgo.cpp

```diff
@@ -222,6 +224,89 @@ void bookmarks( const Node *node, std::vector<NodePtr> &bookmarks )
 	}
 }
 
+void setNumericBookmark( Node *node, int value )
+{
+	if( value < 0 )
+	{
+		return;
+	}
+
+	if( readOnly( node ) )
+	{
+		return;
+	}
+
+	// \todo: should this be const (and the `root` arg in nodesWithMetadata as well?)
+	ScriptNode *scriptNode = node->ancestor<ScriptNode>();
+	if( !scriptNode )
+	{
+		return;
+	}
+
+	for( Node *nodeWithMetadata : Metadata::nodesWithMetadata( scriptNode, g_numericBookmarkName, /* instanceOnly = */ true ) )
+	{
+		if( getNumericBookmark( nodeWithMetadata ) != value )
+		{
+			continue;
+		}
+
+		if( readOnly( nodeWithMetadata ) )
+		{
+			continue;
+		}
+
+		Metadata::deregisterValue( nodeWithMetadata, g_numericBookmarkName );
+	}
+
+	Metadata::registerValue( node, g_numericBookmarkName, new IntData( value ), false );
+}
+
+int getNumericBookmark( const Node *node )
+{
+	ConstIntDataPtr i = Metadata::value<IntData>( node, g_numericBookmarkName );
+	if( !i || readOnly( node ) )
```

Comment:
Here I definitely think we need `ancestorMakesChildNodesReadOnly()`. I should be able to assign a bookmark to a node, then lock it via the NodeEditor and still be able to query the bookmark. But seeing these implementations, I am leaning more heavily towards my suggested alternative API above.